trie/trie_test.go

```diff
@@ -93,3 +93,27 @@ func TestPutAndGet(t *testing.T) {
 		}
 	}
 }
+
+func TestDelete(t *testing.T) {
```

Comment:
I don't think these tests are sufficient. Try the following and see if you can improve the coverage:
```
go test -v ./... -coverprofile=coverage.out
go tool cover -html=coverage.out
```

Reply:
any ideas on how to improve coverage?

Reply:
You need to construct a larger tree and try deleting different parts from it. You are only ever deleting from an extension not a branch. You aren't testing invalid nodes, nor the case where `deleteFromExtension()` doesn't find the key immediately. Pretty sure the current test setup tests the exact same thing 100 times... You might consider constructing some more dynamic test tools, and checkout what the go-ethereum gang does.

Reply:
okay will do that. I think I am deleting from both branches and extensions but yeah I will try to make different testing tools

Reply:
@ansermino I have improved the testing, pls let me know what you think